cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -93,3 +93,75 @@ def parse_run_info_file(self) -> BclConvertRunInfo:
         LOG.info(f"Parsing Run info XML {self.run_info_path}")
         parsed_metrics = BclConvertRunInfo(tree=ET.parse(self.run_info_path))
         return parsed_metrics
+
+    def get_sample_internal_ids(self) -> List[str]:
+        """Return a list of sample internal ids."""
+        sample_internal_ids: List[str] = []
+        for sample_demux_metric in self.demux_metrics:
+            if sample_demux_metric.sample_project not in ["indexcheck", "Undetermined"]:
+                sample_internal_ids.append(sample_demux_metric.sample_internal_id)
+        return sample_internal_ids
+
+    def get_lanes_for_sample_internal_id(self, sample_internal_id: str) -> List[int]:
+        """Return a list of lanes for a sample."""
+        lanes_for_sample: List[int] = []
+        for sample_demux_metric in self.demux_metrics:
+            if sample_demux_metric.sample_internal_id == sample_internal_id:
+                lanes_for_sample.append(sample_demux_metric.lane)
+        return lanes_for_sample
+
+    def get_metrics_for_sample_internal_id_and_lane(
+        self,
+        metrics_list: List[
+            Union[BclConvertQualityMetrics, BclConvertDemuxMetrics, BclConvertAdapterMetrics]
+        ],
+        sample_internal_id: str,
+        lane: int,
+    ) -> Union[BclConvertQualityMetrics, BclConvertDemuxMetrics, BclConvertAdapterMetrics]:
+        """Return the metrics for a sample by sample internal id."""
+        for metric in metrics_list:
+            if metric.sample_internal_id == sample_internal_id and metric.lane == lane:
```

Comment:
There should only be one entry for a sample and lane? At least that is assumed here and I guess it makes sense!

Reply:
Yes the other sample entry is the other lane